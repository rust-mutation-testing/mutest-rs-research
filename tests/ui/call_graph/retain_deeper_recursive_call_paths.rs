//@ print-call-graph
//@ print-targets
//@ stdout
//@ stderr: empty

fn recursive_function(i: usize) {
    if i == 1 { return; }
    recursive_function(i - 1);
}

#[test]
fn test_direct() {
    recursive_function(3);
}

fn indirect_recursive() {
    recursive_function(2);
}

#[test]
fn test_indirect() {
    indirect_recursive();
}
