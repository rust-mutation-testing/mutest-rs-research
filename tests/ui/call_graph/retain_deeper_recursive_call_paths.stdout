
@@@ call graph @@@

entry points:

test test_direct
  -> recursive_function at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:6:1: 6:32 (#0)
test test_indirect
  -> indirect_recursive at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:16:1: 16:24 (#0)

nested calls at distance 1:

recursive_function at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:6:1: 6:32 (#0)
  -> recursive_function at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:6:1: 6:32 (#0)
indirect_recursive at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:16:1: 16:24 (#0)
  -> recursive_function at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:6:1: 6:32 (#0)

@@@ targets @@@

tests -(0)-> recursive_function at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:6:1: 6:32 (#0)
  (0) test_direct
  (1) test_indirect

tests -(0)-> indirect_recursive at tests/ui/call_graph/retain_deeper_recursive_call_paths.rs:16:1: 16:24 (#0)
  (0) test_indirect

targets: 2 total; 2 safe; 0 unsafe (0 tainted)
